packages/protocol/contracts/L1/verifiers/SgxVerifier.sol

```diff
@@ -177,13 +204,16 @@ contract SgxVerifier is EssentialContract, IVerifier {
     }
 
     function _replaceInstance(uint256 id, address oldInstance, address newInstance) private {
+        // Replacing an instance means, it went through a cooldown (if added by on-chain RA) so no
+        // need to have a cooldown
         instances[id] = Instance(newInstance, uint64(block.timestamp));
         emit InstanceAdded(id, newInstance, oldInstance, block.timestamp);
     }
 
     function _isInstanceValid(uint256 id, address instance) private view returns (bool) {
         if (instance == address(0)) return false;
         if (instance != instances[id].addr) return false;
-        return instances[id].addedAt + INSTANCE_EXPIRY > block.timestamp;
+        if (instances[id].validSince > block.timestamp) return false;
+        return instances[id].validSince + INSTANCE_EXPIRY > block.timestamp;
```

Comment:
```suggestion
        return instances[id].validSince <= block.timestamp && block.timestamp <= instances[id].validSince + INSTANCE_EXPIRY;
```

This is really just a suggestion, and I know it's exactly the same, but my brain has trouble with these checks if they are not in the order that make sense to me. And the current code is especially tricky because the first part of the check needs to result to `true` when it's invalid, while the second part needs to result in `true` when it is valid!